src/Nethermind/Nethermind.State/IWorldState.cs

```diff
@@ -1,22 +1,33 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
 
 namespace Nethermind.State
 {
     /// <summary>
     /// Represents state that can be anchored at specific state root, snapshot, committed, reverted.
     /// Current format is an intermittent form on the way to a better state management.
     /// </summary>
-    public interface IWorldState : IJournal<Snapshot>
+    public interface IWorldState : IJournal<Snapshot>, IStateProvider
```

Comment:
Will there be any `IStateProvider` that is not an `IWorldState`? If yes, what will be the case? If no, shall `IStateProvider` be removed and flattened with the world state?